google/generativeai/retriever.py

```diff
@@ -62,18 +59,18 @@ def create_corpus(
     if client is None:
         client = get_default_retriever_client()
 
-    if not name and not display_name:
-        raise ValueError("Either the corpus name or display name must be specified.")
-
     corpus = None
-    if name:
-        if re.match(_CORPORA_NAME_REGEX, name):
-            corpus = glm.Corpus(name=name, display_name=display_name)
-        elif "corpora/" not in name:
-            corpus_name = "corpora/" + re.sub(_PATTERN, "", name)
-            corpus = glm.Corpus(name=corpus_name, display_name=display_name)
-        else:
-            raise ValueError("Corpus name must be formatted as corpora/<corpus_name>.")
+    if re.match(_VALID_NAME, name) and len(name) < 40:
+        corpus_name = "corpora/" + name  # Construct the name
+        corpus = glm.Corpus(name=corpus_name, display_name=display_name)
+    else:
+        raise ValueError(
+            """
```

Comment:
Nit - for convention's sake, make sure the first line of the error message contains something useful - e.g. a short description of the error.

I would also remove the left-padding too, e.g. with textwrap.dedent, so users don't see errors with unnecessary whitespace.